tfhe/src/integer/ciphertext/compressed_modulus_switched_ciphertext.rs

```diff
@@ -0,0 +1,97 @@
+use super::{
+    BaseRadixCiphertext, BaseSignedRadixCiphertext, RadixCiphertext, SignedRadixCiphertext,
+};
+use crate::integer::ServerKey;
+use crate::shortint::ciphertext::CompressedModulusSwitchedCiphertext;
+
+/// An object to store a ciphertext in little memory.
+/// Decompressing it requires a PBS
+///
+/// # Example
+///
+/// ```rust
+/// use tfhe::integer::gen_keys_radix;
+/// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+/// use tfhe::shortint::PBSParameters;
+///
+/// // We have 4 * 2 = 8 bits of message
+/// let size = 4;
+/// let (cks, sks) = gen_keys_radix::<PBSParameters>(PARAM_MESSAGE_2_CARRY_2_KS_PBS.into(), size);
+///
+/// let clear = 3u8;
+///
+/// let ctxt = cks.encrypt(clear);
+///
+/// let compressed_ct = sks.switch_modulus_and_compress(&ctxt);
+///
+/// let decompressed_ct = sks.decompress(&compressed_ct);
+///
+/// let dec = cks.decrypt(&decompressed_ct);
+///
+/// assert_eq!(clear, dec);
+/// ```
+pub type CompressedModulusSwitchedRadixCiphertext =
+    BaseRadixCiphertext<CompressedModulusSwitchedCiphertext>;
+
+impl ServerKey {
+    // Packs a ciphertext to have a smaller serialization size
+    pub fn switch_modulus_and_compress(
+        &self,
+        ct: &RadixCiphertext,
+    ) -> CompressedModulusSwitchedRadixCiphertext {
+        let blocks = ct
+            .blocks
+            .iter()
+            .map(|a| self.key.switch_modulus_and_compress(a))
+            .collect();
+
+        BaseRadixCiphertext { blocks }
+    }
+    // Unpacks a compressed ciphertext
+    // This operation costs a PBS
+    pub fn decompress(
+        &self,
+        compressed_ct: &CompressedModulusSwitchedRadixCiphertext,
+    ) -> RadixCiphertext {
+        let blocks = compressed_ct
+            .blocks
+            .iter()
+            .map(|a| self.key.decompress(a))
+            .collect();
+
+        BaseRadixCiphertext { blocks }
+    }
+}
+
+pub type CompressedModulusSwitchedSignedRadixCiphertext =
+    BaseSignedRadixCiphertext<CompressedModulusSwitchedCiphertext>;
+
+impl ServerKey {
+    // Packs a ciphertext to have a smaller serialization size
+    pub fn switch_modulus_and_compress_signed(
+        &self,
+        ct: &SignedRadixCiphertext,
+    ) -> CompressedModulusSwitchedSignedRadixCiphertext {
+        let blocks = ct
+            .blocks
+            .iter()
+            .map(|a| self.key.switch_modulus_and_compress(a))
+            .collect();
```

Comment:
parallelization as KS may be costly done sequentially

can be a separate function

Reply:
Made parallelized